Rubocop config tune up

Gabe-Torres opened the pull request:
### What github issue is this PR for, if any?
Resolves #6380 

### What changed, and _why_?

Update rubocop config file to specify "plugins" instead of "require" for the following extensions:
- rubocop-capybara
- rubocop-rspec
- rubocop-rspec_rails

This ensures all extensions are supported properly.

### How is this **tested**? (please write rspec and jest tests!) 💖💪

Manual testing 
run `bundle exec standardrb`

We now receive the following output -

```yaml
Warning: Lint/UselessAccessModifier does not support ContextCreatingMethods parameter.

Supported parameters are:

  - Enabled
  ```
  
  Before we received
  
  ```yaml
  Warning: Lint/UselessAccessModifier does not support ContextCreatingMethods parameter.

Supported parameters are:

  - Enabled
rubocop-capybara extension supports plugin, specify `plugins: rubocop-capybara` instead of `require: rubocop-capybara` in /home/runner/work/casa/casa/.rubocop.yml.
For more information, see https://docs.rubocop.org/rubocop/plugin_migration_guide.html.
rubocop-rspec extension supports plugin, specify `plugins: rubocop-rspec` instead of `require: rubocop-rspec` in /home/runner/work/casa/casa/.rubocop.yml.
For more information, see https://docs.rubocop.org/rubocop/plugin_migration_guide.html.
rubocop-rspec_rails extension supports plugin, specify `plugins: rubocop-rspec_rails` instead of `require: rubocop-rspec_rails` in /home/runner/work/casa/casa/.rubocop.yml.
For more information, see https://docs.rubocop.org/rubocop/plugin_migration_guide.html.
WARNING: this project is being migrated to standard gradually via `/home/runner/work/casa/casa/.standard_todo.yml` and is ignoring these files:
  app/controllers/all_casa_admins/sessions_controller.rb
  app/controllers/api/v1/base_controller.rb

  ``` 
  
![CGT](https://github.com/user-attachments/assets/9ee3abcf-dbbc-4451-8331-7089fedf1194)

Gabe-Torres committed:
update: rubocop to specify plugins instead of require for the following extensions

- rubocop-capybara
- rubocop-rspec
- rubocop-rspec_rails

compwron commented:
Good draft! Make sure that the lint part of CI passes. 
(the docker part of CI is temporarily broken in general and not related to this PR)

Gabe-Torres committed:
update: spec test to comply with rubocop inspect and bin/lint command

Following spec files were updated

- spec/system/static/index_spec.rb
- spec/system/volunteers/index_spec.rb
- spec/views/all_casa_admins/patch_notes/index.html.erb_spec.rb

Gabe-Torres commented:
> Good draft! Make sure that the lint part of CI passes. (the docker part of CI is temporarily broken in general and not related to this PR)

Gotcha! All checks passing now 👍🏽 Looks like main was recently updated so I can rebase/merge when approved. Thank you!
